llama_stack/providers/inline/post_training/torchtune/utils.py

```diff
@@ -0,0 +1,110 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+
+# Copyright (c) Meta Platforms, IAny, nc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+
+from enum import Enum
+from typing import Any, Callable, Dict, List
+
+import torch
+from llama_stack.apis.datasets import Datasets
+from llama_stack.apis.common.type_system import *  # noqa
+from llama_models.sku_list import resolve_model
+from llama_stack.apis.common.type_system import ParamType
+
+from torchtune.models.llama3 import llama3_tokenizer, lora_llama3_8b
+from torchtune.models.llama3._tokenizer import Llama3Tokenizer
+from torchtune.models.llama3_2 import lora_llama3_2_3b
+
+
+class ColumnName(Enum):
+    instruction = "instruction"
+    input = "input"
+    output = "output"
+    text = "text"
+
+
+MODEL_CONFIGS: Dict[str, Dict[str, Any]] = {
+    "Llama3.2-3B-Instruct": {
+        "model_definition": lora_llama3_2_3b,
+        "tokenizer_type": llama3_tokenizer,
+        "checkpoint_type": "LLAMA3_2",
+    },
+    "Llama-3-8B-Instruct": {
+        "model_definition": lora_llama3_8b,
+        "tokenizer_type": llama3_tokenizer,
+        "checkpoint_type": "LLAMA3",
+    },
+}
+
+EXPECTED_DATASET_SCHEMA: Dict[str, List[Dict[str, ParamType]]] = {
+    "alpaca": [
```

Comment:
A few questions:

1. what do these three options mean? 
2. what does `instruction` mean? does it mean `system_prompt`? 
3. do you think we can use the types we have in the rest of our system -- for example, how is a dialog represented? We should be able to re-use the `UserMessage`, `SystemMessage` types we have in the rest of the system. Evals uses some of them.

Reply:
> what do these three options mean?

the 3 options mean 3 eligible alpaca dataset schema. 'input' and 'text' columns are optional for alpaca dataset schema (see: https://github.com/pytorch/torchtune/blob/9cfa28835246a4c1ac4449e703eae8f49227db55/torchtune/data/_messages.py#L696 and https://huggingface.co/datasets/tatsu-lab/alpaca?row=0).

 > what does instruction mean? does it mean system_prompt?
 
instruction is different from system_prompt here. In alpaca dataset, 'instruction' pairs with 'input' as user_prompt (example: https://github.com/pytorch/torchtune/blob/9cfa28835246a4c1ac4449e703eae8f49227db55/torchtune/data/_messages.py#L696)

>  do you think we can use the types we have in the rest of our system -- for example, how is a dialog represented? We should be able to re-use the UserMessage, SystemMessage types we have in the rest of the system. Evals uses some of them.

torchtune has its own Message definition in the data transform https://github.com/pytorch/torchtune/blob/9cfa28835246a4c1ac4449e703eae8f49227db55/torchtune/data/_messages.py#L724. I lean toward directly import torchtune data transform to the stack and reuse its Message type. For dataset schema validation, I refer to how eval does https://github.com/meta-llama/llama-stack/blob/41487e6ed143a3acb72fe331da41df4ad5cdb2cb/llama_stack/providers/inline/eval/meta_reference/eval.py#L80